src/main/java/im/status/keycard/KeycardApplet.java

```diff
@@ -223,6 +238,22 @@ public KeycardApplet(byte[] bArray, short bOffset, byte bLength) {
    * @throws ISOException any processing error
    */
   public void process(APDU apdu) throws ISOException {
+    byte[] apduBuffer = apdu.getBuffer();
```

Comment:
Instead of adding these if statement checks, use the switch case below, this split in the logic path adds redundancy and complexity.

Reply:
Also remove cert specific logic from here and put them in their own methods.

Reply:
As we just discussed, `processInit` blocks the desired functionality and unfortunately we need to shift around some of the steps. I also just verified that the tests all pass.